spec/models/course/material_spec.rb

```diff
@@ -30,5 +31,52 @@
         expect(subject).to be_invalid
       end
     end
+
+    describe 'abilities' do
+      subject(:ability) { Ability.new(user) }
+      let(:course) { create(:course) }
+      let(:valid_material) do
+        folder = build_stubbed(:folder, course: course)
+        build_stubbed(:material, folder: folder)
+      end
+      let(:not_started_material) do
+        folder = build_stubbed(:folder, :not_started, course: course)
+        build_stubbed(:material, folder: folder)
+      end
+      let(:ended_material) do
+        folder = build_stubbed(:folder, :ended, course: course)
+        build_stubbed(:material, folder: folder)
+      end
+      let(:started_linked_material) do
+        folder = build_stubbed(:folder, course: course,
+                                        owner: build_stubbed(:course_assessment_category))
+        build_stubbed(:material, folder: folder)
+      end
+      let(:not_started_linked_material) do
+        folder = build_stubbed(:folder, :not_started,
+                               course: course, owner: build_stubbed(:course_assessment_category))
+        build_stubbed(:material, folder: folder)
+      end
+
+      context 'when the user is a Course Student' do
```

Comment:
Would it be better to have a course_material_abilities_spec.rb together with the models?